src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs

```diff
@@ -96,13 +101,20 @@ public static byte[] CreateSigningKeyMaterial(string userSecretsId, string schem
         }
 
         secrets ??= new JsonObject();
-        var key = GetSigningKeyPropertyName(schemeName, issuer);
+        var signingKeyValueName = GetSigningKeyValuePropertyName(schemeName);
+        var signingKeyLengthName = GetSigningKeyLengthPropertyName(schemeName);
 
-        if (reset && secrets.ContainsKey(key))
+        if (reset && secrets.ContainsKey(signingKeyValueName))
         {
-            secrets.Remove(key);
+            secrets.Remove(signingKeyValueName);
         }
-        secrets.Add(key, JsonValue.Create(Convert.ToBase64String(newKeyMaterial)));
+        if (reset && secrets.ContainsKey(signingKeyLengthName))
+        {
+            secrets.Remove(signingKeyLengthName);
+        }
+
+        secrets.Add(signingKeyValueName, JsonValue.Create(Convert.ToBase64String(newKeyMaterial)));
```

Comment:
If a misconfigured signing key was found and we then go into this method to create a new key, this `secrets.Add` call would noop (or maybe throw) because `reset` wouldn't be set and we wouldn't remove the old key.